### ISSUE TITLE
Test suite fails if before contains an Allure step

### ISSUE THREAD

**Issue by Yethal** [Shelex/cypress-allure-plugin#11]
**Describe the bug**
If before contains an Allure step (via cy.allure() chainer) test suite fails with following error message:
```
Cannot read property 'startStep' of null
```
**To Reproduce**
1. Create a test suite with before section
2. Add an Allure step to before section (either directly or via custom command that has step definition embedded)
3. Run the test suite

**Expected behavior**
Test suite runs without errors. Steps contained in before section are shown on setUp of every test case

**Environment (please complete the following information):**
- Cypress version:                                4.7.0
- OS:                                                     Windows|Linux (CentOS)

** Developer Tools Console Output **

**Additional context**

**Comment by Shelex** [comment]
Hello @Yethal 
Thank you for opening an issue.

As I see there is actually 2 issues:
1. `before` and `after` hooks are not logged into allure.
Usually these hooks contain some pre-requisites which have little connection to tests.
Moreover, suggestion to have them `are shown on setUp of every test case` is quite misleading as they happen only once per suite.
2. allure api used in `before` or `after` causes error.
This definitely need to be addressed by improving error message.

Or could you please provide some use case of having `before` and `after` hooks in report with used allure api?

**Comment by Yethal** [comment]
Most of my test cases require a user to log in first. I could have the tests log in every time a test case starts (and move the login part to beforeEach) but it's more efficient to log in once and then run multiple test cases in a single user session. That works fine for me except the Allure report doesn't show up the steps that lead to the user accessing the application as those happen within the before hook

**Comment by Shelex** [comment]
So basically it is feature request to have `before` and `after` hooks to be logged into allure and be duplicated for every test.

**Comment by Yethal** [comment]
And make it configurable (if possible), I assume there are a lot of people who like the way the plugin works now. 
As for the allure api causing errors in `before` and `after`, it would be easier if it was documented which calls can be used in which sections of the suite. My initial assumption for example was that calling allure().story or allure().feature in `before` hook would apply that data to every test case in the suite, instead it needs to be called in `beforeEach`. That behavior is totally fine of course, it's just a matter of communicating that to the user

**Comment by archital** [comment]
> Or could you please provide some use case of having before and after hooks in report with used allure api?

In my case I want to use .stepStart() - .stepEnd() as a wrapper above the Cypress custom command. But the custom command could be executed before()/beforeAll() as a setup arrange and also in the test itself as an action.

**Comment by Shelex** [comment]
@archital I have not got your case, as custom commands are also logged for now.
As for the current state (since 1.7.0), steps implementation is revisited and now it should be available inside hooks also. However, implementing showing `before` and `after` hooks will be done in terms of separate task, as this one is about errors thrown.
